### PR TITLE
feat: add source maps

### PR DESCRIPTION
**Pull request by dsseng**
Add source maps to built scripts and styles. This will help debugging. Should be done for #3 after it merge

### CODE REVIEW

**Comment by dsseng** [comment]
@yyx990803 could you please review?

File: middleware.js [hunk 1]
```diff
@@ -31,6 +31,30 @@ const vueMiddleware = (options = defaultOptions) => {
...
+    )
+
+    return {
+      ...script,
+      code: `//# sourceMappingURL=data:application/json;base64,${map}\n` + script.code
+    }
+  }
+
+  function injectSourceMapsToStyles (styles) {
+    return styles.map(s => {
```
Comment (yyx990803) [inline middleware.js]: Isn't the logic duplicated with `injectSourceMapsToScript` here?

Comment (yyx990803) [reply middleware.js]: Can just rename `injectSourceMapsToScript` and reuse it here

**Review by yyx990803 — CHANGES_REQUESTED** [review]

File: middleware.js [hunk 2]
```diff
@@ -54,13 +78,26 @@ const vueMiddleware = (options = defaultOptions) => {
...
+      ...descriptorResult,
+      script: injectSourceMapsToScript(descriptorResult.script),
+      styles: injectSourceMapsToStyles(descriptorResult.styles)
+    })
+    return { ...assembledResult, updateTime }
   }
 
   return async (req, res, next) => {
     if (req.path.endsWith('.vue')) {
+      const { filepath, source } = await readSource(req)
```
Comment (yyx990803) [inline middleware.js]: Should use `bundleSFC` here?

**Review by yyx990803 — CHANGES_REQUESTED** [review]

File: middleware.js (hunk 1 above)
Comment (dsseng) [reply middleware.js]: @yyx990803 JS comment is `//`, CSS one is `/* */`

File: middleware.js (hunk 2 above)
Comment (dsseng) [reply middleware.js]: yes, thanks!

Comment (dsseng) [reply middleware.js]: @yyx990803 fixed, please review

**Comment by dsseng** [comment]
CSS mapping demo:
default

File: middleware.js (hunk 1 above)
Comment (dsseng) [reply middleware.js]: but maybe CSS-style will work in JS too :thinking:

Comment (dsseng) [reply middleware.js]: no, it doesn't, but we can add something like `commStyle` parameter to our fn.

Comment (dsseng) [reply middleware.js]: @yyx990803 I've done some refactor, please review

**Pull request merged**